protocol/src/main.rs

```diff
@@ -0,0 +1,234 @@
+// Copyright 2023 Oxide Computer Company
+
+use std::net::SocketAddr;
+use std::time::Duration;
+use std::time::Instant;
+
+use crucible_common::Block;
+use crucible_protocol::BlockContext;
+use crucible_protocol::CrucibleDecoder;
+use crucible_protocol::CrucibleEncoder;
+use crucible_protocol::EncryptionContext;
+use crucible_protocol::JobId;
+use crucible_protocol::Message;
+
+use anyhow::Result;
+use clap::Parser;
+use futures::SinkExt;
+use futures::StreamExt;
+use tokio::net::TcpListener;
+use tokio::net::TcpSocket;
+use tokio_util::codec::FramedRead;
+use tokio_util::codec::FramedWrite;
+use uuid::Uuid;
+
+#[derive(Debug, Parser)]
+#[clap(about = "Protocol serialization and deserialization speed test")]
+enum Args {
+    Dynamometer {
+        #[clap(short, long, default_value_t = 512)]
+        write_size: usize,
+
+        /// Number of write protocol messages to submit at one time
+        #[clap(short, long, default_value_t = 1)]
+        num_writes: usize,
+
+        /// Number of write requests per write protocol message
+        #[clap(short, long, default_value_t = 1)]
+        requests_per_write: usize,
+
+        /// Number of samples to exit for
+        #[clap(short, long, default_value_t = 10)]
+        samples: usize,
+    },
+}
+
+#[tokio::main]
```

Comment:
I don't love that the propocol lib is now a binary.  It might be a little confusing
to someone coming in.  Could we maybe rename the resulting binary `protocol_test` or
`ptest` or something to indicate that?

An additional idea, maybe a README somewhere talking about it.  I've been negligent
as well in documenting what all the stuff in the crucible repo is, and we should start
getting better about leaving breadcrumbs behind to assist new readers on what the
heck all this stuff is.

Reply:
Agreed, I'd even suggest a separate binary crate so that `protocol` remains a pure library crate.